### DISCUSSION TITLE
Lighthouse Expansion: Spotty Receiving, calibration

### DISCUSSION THREAD

**Discussion by HawkMock** [bitcraze/discussions#924] [Q&A] [answered]
I am setting up an area to be able to use the Lighthouse expansion with the Steam v2 base-stations. When I put the drone is the area where I should be receiving sweep angles, only one or two of the "receiving" boxes in cfclient will flash green, and only some of the boxes will light up with green, which then turns yellow and then blue/greyish, saying the calibration is "cached". The area I am using is a large, empty square room 10 meters by 10 meters wide, and the base-stations are mounted about 3 meters off the ground at the corners of the room. I have 4 stations. The area seems the rights size, I would expect to have good reception. Any ideas what might be going on?

**Comment by krichardsson** [comment]
> The area I am using is a large, empty square room 10 meters by 10 meters wide, and the base-stations are mounted about 3 meters off the ground at the corners of the room. I have 4 stations. The area seems the rights size, I would expect to have good reception.

10 x 10 sounds pretty large and like you are on the edge. The absolute max range of the base stations are 6 meters. If your base station is at 3 m height and in the center of the side you get sqrt(5*5 + 3*3) = 5.8 meters to the center of the floor. Should work, but with very little margin :-)
You also have to consider the angle to the sensors from the base stations, in the center of the space (which would be your worst case) the highest possible height will be limited as the sensors will not receive any light as the base stations is more on the side than above.

> When I put the drone is the area where I should be receiving sweep angles, only one or two of the "receiving" boxes in cfclient will flash green

My guess is that this is mainly due to the size of the space. There is also a [known bug](https://github.com/bitcraze/lighthouse-fpga/issues/14) that prevents reception of base stations in some angles, but this is probably not what you see.   

It is enough to receive one or two base stations for flight and it might work out, the best way is to try. 

> which then turns yellow and then blue/greyish, saying the calibration is "cached"

The calibration data status is [documented here](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/lighthouse_tab/#base-station-status). The calibration data is transferred in the light beams from the base stations, but since this takes a while, they are stored in persistent memory to be able to take off quickly after a reboot. Blue means that the cached data is used, green that data has been received through the light beams. Blue or green are both fine and not important unless you swap a base station for a new one.

**Reply by HawkMock** [reply]
Thank you for your reply. I've tinkered around a bit more, and I've got some better measurements of my room. My room is actually closer to 6mx6mx3m, so it should be within the range of 6m as we talked about earlier as the centered floor position is 5.2 meters away rather than 5.8. The basestations are angled downward about 35deg from horizontal. Still, when I place the drone at the centered floor position, I get basically no reception-- one of the receiving squares in cfclient might blink for a quarter second every minute or two. Any other thoughts on where I can start troubleshooting?

**Reply by HawkMock** [reply]
Some additonal info, my X, Y, and Z coordinates for my drone in the lighthouse positioning window increase or decrease linearly and the blue drone dot will move in a line in the coordinate grid in random directions. Not sure if this is related to my issue or the expected behavior for an uncalibrated system.

**Comment by krichardsson** [comment]
6x6x3 should work, this is very similar to our flight space. 

My suggestion is to turn 3 of the base stations off and start with only one active, to see if that makes any difference. Is it possible to receive this? How far away? Does it receive the calibration data (the "calibration" indicator turn green)?

If/when this works turn on a second base station and check if it works, and so on. 

It is possible that there is a problem with the deck (let's hope not). If you happen to have more than one lighthouse deck, you could try with that to see if there is any difference.

> Some additonal info, my X, Y, and Z coordinates for my drone in the lighthouse positioning window increase or decrease linearly and the blue drone dot will move in a line in the coordinate grid in random directions. Not sure if this is related to my issue or the expected behavior for an uncalibrated system.

This is most likely just the kalman estimator that is confused as it is fed "weird", or no data. The estimated position typically slides away in some random direction until the estimator automatically resets. It will not do anything useful until the lighthouse system is set up properly.

**Reply by HawkMock** [reply, marked as answer]
Thanks again for the response.

It turns out my issue was where I had the board mounted. According to [the page on expansion deck orientations](https://www.bitcraze.io/documentation/tutorials/getting-started-with-expansion-decks/), the third diagram shows you can use a batter holder and battery on top with the expansion board on the bottom, which is what I had.

I ended up essentially reversing this-- I pushed the pins through and mounted the battery on the bottom and mounted the lighthouse deck on top. Immediately upon doing this, I received signal as expected, losing reception only when moving outside the 6m distance from any given base station. I believe the battery and main board were blocking the reception from the lighthouses. I don't remember reading anywhere that the board must be mounted on top.

The reason I put the battery underneath the main board rather than between the lighthouse board and main board is to lower the center of gravity and increase stability.

**Reply by knmcguire** [reply]
Hi! Great that you found out what was wrong. It is important that the lighthouse deck is all the way to the top as it is crucial that nothing is blocking the lighthouse basestations IR planes.
